### PR TITLE
[nrfconnect] Added on-boarding codes generation to the factory data script

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Generate onboarding parameters:
 - manual pairing code
 - QRCode using Nrfconnect factory data script.

To generate onboarding parameters add --generate_onboarding to the generate_nrfconnect_chip_factory_data.py script.

### CODE REVIEW

File: docs/guides/nrfconnect_factory_data_configuration.md [hunk 1]
```diff
@@ -493,6 +493,45 @@ as an additional argument. To do this, complete the following steps:
...
+3. Add the `--generate_onboarding` argument to the Python script final
+   invocation:
+
+    ```
+    $ python scripts/tools/nrfconnect/generate_nrfconnect_chip_factory_data.py ...<arguments>... --generate_onboarding
+    ```
+
+4. Run the script.
+
+5. Navigate to the output directory provided as `-o` argument.
```
Comment (greg-fer) [inline docs/guides/nrfconnect_factory_data_configuration.md:526]:
```suggestion
5. Navigate to the output directory provided as the `-o` argument.
```

File: docs/guides/nrfconnect_factory_data_configuration.md [hunk 2]
```diff
@@ -493,6 +493,45 @@ as an additional argument. To do this, complete the following steps:
...
+
+6. Look for three files:
+
+    - .json file containing the latest factory data set.
+    - .txt file containing the generated manual code and QRCode in a text
+      version.
+    - .png file containing the generated QRCode as an image.
```
Comment (greg-fer) [inline docs/guides/nrfconnect_factory_data_configuration.md:533]:
```suggestion
The output directory contains the following files you need:

- JSON file containing the latest factory data set.
- Test file containing the generated manual code and the text version of the QR Code.
- PNG file containing the generated QR Code as an image.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: .github/workflows/examples-nrfconnect.yaml [hunk 3]
```diff
@@ -99,7 +99,9 @@ jobs:
               run: scripts/run_in_build_env.sh "python3 scripts/setup/nrfconnect/update_ncs.py --check"
             - name: Run unit tests of factory data generation script
               timeout-minutes: 15
-              run: scripts/run_in_build_env.sh "./scripts/tools/nrfconnect/tests/test_generate_factory_data.py"
+              run: |
+                  scripts/run_in_build_env.sh 'pip3 install -r scripts/setup/requirements.nrfconnect.txt'
```
Comment (bzbarsky-apple) [inline .github/workflows/examples-nrfconnect.yaml:103]: I wonder whether it's worth having this as part of the Docker image...  followup for that if so.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: docs/guides/nrfconnect_factory_data_configuration.md [hunk 4]
```diff
@@ -493,6 +495,59 @@ as an additional argument. To do this, complete the following steps:
...
+    QR Code.
+-   PNG file containing the generated QR Code as an image.
+
+#### Enabling onboarding codes generation within the build system
+
+You can generate onboarding codes using the nRF Connect platform build system
+described in
+[Building an example with factory data](#building-an-example-with-factory-data),
+and build an example with the following additional option:
+`-DCONFIG_CHIP_FACTORY_DATA_GENERATE_ONBOARDING_CODES=y`:
```
Comment (greg-fer) [inline docs/guides/nrfconnect_factory_data_configuration.md:540]:
```suggestion
`-DCONFIG_CHIP_FACTORY_DATA_GENERATE_ONBOARDING_CODES=y`.
```

**Review by greg-fer — APPROVED** [review]
One nit.

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**